### PR TITLE
Null safety migration

### PR DESCRIPTION
**Pull request by natebosch**
Migrates code and tests. Passes static analysis.

### CODE REVIEW

File: lib/src/multi_channel.dart [hunk 1]
```diff
@@ -132,15 +132,15 @@ class _MultiChannel<T> extends StreamChannelMixin<T>
   /// it's coming from a channel that was originally created locally.
   var _nextId = 1;
 
-  _MultiChannel(this._inner) {
+  _MultiChannel(StreamChannel<dynamic> inner) : _inner = inner {
```
Comment (jakemac53) [inline lib/src/multi_channel.dart]: hah, that's an interesting one

File: pubspec.yaml [hunk 2]
```diff
@@ -8,11 +8,17 @@ author: Dart Team <misc@dartlang.org>
...
+  sdk: '>=2.8.0-dev.15 <3.0.0'
 
 dependencies:
   async: '>=1.11.0 <3.0.0'
 
 dev_dependencies:
   pedantic: ^1.8.0
   test: ^1.2.0
+
+dependency_overrides:
```
Comment (jakemac53) [inline pubspec.yaml]: You might want to add the full list of overrides for the tests, https://gist.githubusercontent.com/jakemac53/23f62a98d6226e80c082264fe9f0e6b3/raw/4a699ae440ee4b1060a05d56011bcfa8d1de701e/overrides.yaml

File: lib/src/multi_channel.dart (hunk 1 above)
Comment (natebosch) [reply lib/src/multi_channel.dart]: Yeah, it's unusual to have a field which always starts with a value then eventually becomes null.

File: pubspec.yaml (hunk 2 above)
Comment (natebosch) [reply pubspec.yaml]: Done

**Comment by natebosch** [comment]
Most tests pass in weak mode, see some warnings in `packag:async`.

The `test/isolate_channel_test.dart` never exits when run this way, but the tests pass.

One test crashes the VM:

```
dart '--enable-experiment=non-nullable' test/with_close_guarantee_test.dart
Crash when compiling package:async/async.dart,
at character offset null:
NoSuchMethodError: The getter 'uri' was called on null.
Receiver: null
```

**Comment by natebosch** [comment]
For the test that crashes, filed https://github.com/dart-lang/sdk/issues/41102 and https://github.com/dart-lang/sdk/issues/41103

**Comment by natebosch** [comment]
For the test that never exits, should be fixed by https://github.com/dart-lang/stream_channel/pull/54

**Pull request merged**